src/components/organisms/WithNavigationBar/WithNavigationBar.tsx

```diff
@@ -16,17 +31,71 @@ export const WithNavigationBar: React.FunctionComponent<PropsType> = ({
   redirectionUrl,
   fullscreen,
   children,
-  hasBackButton,
-}) => (
-  <>
-    {/* @debt remove backButton from Navbar */}
-    <NavBar redirectionUrl={redirectionUrl} hasBackButton={hasBackButton} />
-    <div className={`navbar-margin ${fullscreen ? "fullscreen" : ""}`}>
-      {children}
+}) => {
+  const { user } = useUser();
+  const venueId = useVenueId();
+  const venue = useSelector(currentVenueSelectorData);
+  const navigateToHomepage = useCallback(() => {
+    const venueLink =
+      redirectionUrl ?? venueId ? venueInsideUrl(venueId ?? "") : "/";
+
+    window.location.href = venueLink;
+  }, [redirectionUrl, venueId]);
```

Comment:
Can we use react-router's helpers here instead of `window.location.href`?

Eg.

- https://reactrouter.com/web/api/Hooks/usehistory